apps/indexer/src/api/services/coingecko/coingecko.service.ts

```diff
@@ -37,4 +53,37 @@ export class CoingeckoService {
       });
     }
   }
+
+  async getTokenPrice(
+    tokenId: CoingeckoTokenId,
+    tokenContractAddress: string,
+    targetCurrency: string,
+  ): Promise<number> {
+    try {
+      const assetPlatform = CoingeckoIdToAssetPlatformId[tokenId];
+      const response = await fetch(
+        `${this.coingeckoApiUrl}/simple/token_price/${assetPlatform}?contract_addresses=${tokenContractAddress}&vs_currencies=${targetCurrency}`,
+        {
+          headers: {
+            "x-cg-demo-api-key": this.coingeckoApiKey,
+          },
+        },
+      );
+
+      if (!response.ok) {
+        throw new Error(`HTTP ${response.status}: ${response.statusText}`);
+      }
+
+      const data = await response.json();
+      return createCoingeckoTokenPriceDataSchema(
+        tokenContractAddress.toLowerCase(),
```

Comment:
use getAddress from viem that parses the checksum

```suggestion
        getAddress(tokenContractAddress),
```